specs/AppLifecycle/Instancing/CustomEventArgs.md

```diff
@@ -0,0 +1,170 @@
+
+A page for [Microsoft.Windows.AppLifecycle.ExtendedActivationKind](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.extendedactivationkind) exists.
+
+The following would be added to [Fields](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.extendedactivationkind).
+
+|_Name_|_Contract_|_Description_                       |
+|------|----------|------------------------------------|
+|UserDefined| 1   | `Data` consists of a custom object.|
+
+
+## AppActivationArguments.AddCustomArgs method
+
+This is the class for [AppActivationArguments](https://docs.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.windows.applifecycle.appactivationarguments).  The only change is adding AddCustomArgs
+
+Example:
+
+```c#
+// The rest of AppActivationArguments here.
+
+public void AddCustomArgs(IInspectable newArguments) 
+{
+	m_Kind = Microsoft.Windows.AppLifecycle.ExtendedActivationKind.UserDefined;
+	m_Data = newArguments;
+}
+```
+
+# API Details
+
+```c# (but really MIDL3)
+namespace Microsoft.Windows.AppLifecycle
+{
+    [contractversion(3)]
+    apicontract AppLifecycleContract {};
+
+    [contract(AppLifecycleContract, 1)]
+    enum ExtendedActivationKind
+    {
+        Launch = 0,
+        Search,
+        ShareTarget,
+        File,
+        Protocol,
+        FileOpenPicker,
+        FileSavePicker,
+        CachedFileUpdater,
+        ContactPicker,
+        Device,
+        PrintTaskSettings,
+        CameraSettings,
+        RestrictedLaunch,
+        AppointmentsProvider,
+        Contact,
+        LockScreenCall,
+        VoiceCommand,
+        LockScreen,
+        PickerReturned = 1000,
+        WalletAction,
+        PickFileContinuation,
+        PickSaveFileContinuation,
+        PickFolderContinuation,
+        WebAuthenticationBrokerContinuation,
+        WebAccountProvider,
+        ComponentUI,
+        ProtocolForResults = 1009,
+        ToastNotification,
+        Print3DWorkflow,
+        DialReceiver,
+        DevicePairing,
+        UserDataAccountsProvider,
+        FilePickerExperience,
+        LockScreenComponent,
+        ContactPanel,
+        PrintWorkflowForegroundTask,
+        GameUIProvider,
+        StartupTask,
+        CommandLineLaunch,
+        BarcodeScannerProvider,
+        PrintSupportJobUI,
+        PrintSupportSettingsUI,
+        PhoneCallActivation,
+        VpnForeground,
+        // NOTE: Values below 5000 are designated for the platform.  The above list is kept in sync with
+        // Windows.ApplicationModel.Activation.ActivationKind.
+
+        Push = 5000,
+
+        [contract(AppLifecycleContract, 2)]
+        AppNotification,
+		
+        [contract(AppLifecycleContract, 3)]
+        UserDefined,
+    };
+
+    [contract(AppLifecycleContract, 1)]
+    runtimeclass AppActivationArguments
+    {
+        ExtendedActivationKind Kind { get; };
+        IInspectable Data{ get; };
```

Comment:
In MIDL3, use `Object` instead of `IInspectable` - it projects in the metadata as `Object` to C# and `IInspectable` to C++/WinRT.